runtime/src/main/java/org/corfudb/util/NodeLocator.java

```diff
@@ -83,55 +109,29 @@ public static NodeLocator parseString(String toParse) {
             } else {
                 String[] query = url.getQuery().split("&");
                 options = Arrays.stream(query)
-                    .map(keyValue -> keyValue.split("="))
-                    .collect(Collectors.toMap(kv -> kv[0], kv -> kv[1]));
+                        .map(keyValue -> keyValue.split("="))
+                        .collect(Collectors.toMap(kv -> kv[0], kv -> kv[1]));
             }
 
             return NodeLocator.builder()
-                            .protocol(proto)
-                            .host(host)
-                            .port(port)
-                            .nodeId(nodeId)
-                            .options(options)
-                            .build();
+                    .protocol(proto)
+                    .host(host)
+                    .port(port)
+                    .nodeId(nodeId)
+                    .options(options)
+                    .build();
 
         } catch (URISyntaxException m) {
             throw new IllegalArgumentException(m);
         }
     }
 
+    public String toEndpointUrl(){
+        return host + ":" + port;
```

Comment:
Why not create this during construction?

Reply:
Don't want to create an extra variable

Reply:
I suppose this shouldn't generate too much garbage.